components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4960,4 +4963,38 @@ public static String[] extractCredentialsFromAuthzHeader(HttpServletRequest requ
 
         return OAuthUtils.decodeClientAuthenticationHeader(authorizationHeader);
     }
+
+    /**
+     * Check whether TLS client certificate bound access tokens are supported by the server.
+     *
+     * @return     Whether TLS client certificate bound access tokens are supported.
+     */
+    public static boolean isTlsClientCertificateBoundAccessTokensSupported() {
+
+        List<OAuthClientAuthenticator> clientAuthenticators = OAuth2ServiceComponentHolder.getAuthenticationHandlers();
+        for (OAuthClientAuthenticator clientAuthenticator : clientAuthenticators) {
+            if (MTLS_CLIENT_AUTHENTICATOR.equals(clientAuthenticator.getName())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Retrieve the list of client authentication methods supported by the server.
+     *
+     * @return     Client authentication methods supported by the server.
+     */
+    public static String[] getSupportedClientAuthMethods() {
+
+        List<OAuthClientAuthenticator> clientAuthenticators = OAuth2ServiceComponentHolder.getAuthenticationHandlers();
+        List<String> supportedClientAuthMethods = new ArrayList<>();
```

Comment:
use a hashset to make sure that no duplicate values gets added.

Reply:
Fixed in 3c54ef90dce0b0bb1fc707a493b621553251eba8